crates/ark/src/help_proxy.rs

```diff
@@ -144,23 +144,29 @@ async fn proxy_request(req: HttpRequest, app_state: web::Data<AppState>) -> Http
     match client.get(target_url.clone()).send().await {
         // OK.
         Ok(response) => {
-            // We only handle OK. Everything else is unexpected.
-            if response.status() != reqwest::StatusCode::OK {
-                return HttpResponse::BadGateway().finish();
-            }
-
             // Get the headers we need.
             let headers = response.headers().clone();
             let content_type = headers.get("content-type");
 
             // Log.
             log::info!(
-                "Proxying URL '{:?}' path '{}' content-type is '{:?}'",
+                "Proxying URL {:?} path '{}' content-type is '{:?}'",
                 target_url.to_string(),
                 target_url.path(),
                 content_type,
             );
 
+            // We only handle OK. Everything else is unexpected.
+            if response.status() != reqwest::StatusCode::OK {
+                log::error!(
+                    "Got status {} proxying {:?}: {:?}",
+                    response.status().clone().to_string(),
+                    target_url.to_string(),
+                    response.text().await.unwrap(),
```

Comment:
Since something is wrong in this code path, can we confidently say that `response` can be properly decoded? Might be better to condition this on `Ok()`?

Reply:
Agreed about the `unwrap()` feeling unsafe here since we are in a `!= reqwest::StatusCode::OK` state here

Reply:
Thanks, both! Improved in a85e59227113d85ea09529c40e294a43c11fe3dc